src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -1463,6 +1544,10 @@ impl Decimal {
         }
         res
     }
+
+    pub fn is_zero(&self) -> bool {
+        !self.word_buf.iter().any(|&x| x != 0)
```

Comment:
word_buf's length is fixed. You should check if all the efficient bytes are zeros. It's useless to reset all bytes to zero.